Source/JSONAPISerializer.swift

```diff
@@ -240,16 +250,27 @@ public extension JSONAPIEntity {
         var attributes = [String: AnyObject]()
         var relationships = [String: AnyObject]()
         
+        if let linkedEntity  = self as? JSONAPILinkedEntity,
+               entityLinks = linkedEntity.links where entityLinks.count > 0 {
+            data["links"] = linkedEntity.links.serialize()
+        }
+        
         for child in mirror.children {
             if let label = child.label {
                 if let value = child.value as? JSONAPISerializable, let data = value.data(includeRelationships: false, includeAttributes: false) {
                     if includeRelationships {
-                        relationships[label] =  ["data": data]
+                        var relationship = [String: AnyObject](dictionaryLiteral: ("data", data))
+                        
+                        if let value = value as? JSONAPILinkedEntity {
+                            relationship["links"] = value.topLinks.serialize()
+                        }
+                        
+                        relationships[label] = relationship
                     }
-                } else if includeAttributes {
+                } else if includeAttributes && !["id", "links", "topLinks"].contains(label)  {
```

Comment:
Nice but let's create a 
let excludedAttributes =....
So we don't create a dict every time and it's more clear